### PR TITLE
Meetup checks now update the target google calendar instead of always adding new events.

### PR DESCRIPTION
**Pull request by kyleswartley**
Added an in-memory database that is used to track events that have been added to the calendar.
- New events since the last check are added to the calendar
- Preexisting events have their details updated to match with the meetup.com event, whether or not there has been a change

cc: issue #14

### CODE REVIEW

**Comment by skapadia** [comment]
Nice use of HSQLDB, although an in-memory database is probably overkill for this project.  It doesn't hurt though.  FYI there's also an in-memory database that's distributed with the Oracle JDK.  Its Apache Derby, branded as "Java DB".

File: src/main/resources/schema.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE id_table(
```
Comment (skapadia) [inline src/main/resources/schema.sql]: Please rename this table to `calendar_event` or `calendar_event_key`.  I prefer the former. Tables should have clear names (i.e. include the name of the entity, etc.).  Although we're not storing any other information about the calendar event here, we may in the future.

File: src/main/java/com/chariotsolutions/meetupnotifier/MeetupNotifierImpl.java [hunk 2]
```diff
@@ -6,41 +6,55 @@
...
-
-      List<Event> events = results.getResults().stream()
-          .map(new GoogleConverter()::convertTo)
-          .collect(Collectors.toList());
-
-      for (Event googleEvent : events) {
-        App.cal.events().insert("primary", googleEvent).execute();
-        logger.debug("An event was added to the calendar successfully");
+      JdbcTemplate template = new JdbcTemplate(ds);
+      String metupInCalendarQuery = "SELECT * from id_table where meetupID=?";
```
Comment (skapadia) [inline src/main/java/com/chariotsolutions/meetupnotifier/MeetupNotifierImpl.java]: Please extract the SQL strings out into private constants.

File: src/main/java/com/chariotsolutions/meetupnotifier/EventIdRow.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package com.chariotsolutions.meetupnotifier;
+
+public class EventIdRow {
```
Comment (skapadia) [inline src/main/java/com/chariotsolutions/meetupnotifier/EventIdRow.java]: Please rename this to `CalendarEvent` or `CalendarEventKey`, depending on what you rename the table to.

**Comment by kyleswartley** [comment]
> probably overkill

Should be this project's subtitle.
I'll look into Derby

File: src/main/java/com/chariotsolutions/meetupnotifier/EventIdRow.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  private String googleId;
+  private String meetupId;
+
+  public EventIdRow(String googleId, String meetupId) {
+    this.googleId = googleId;
```
Comment (skapadia) [inline src/main/java/com/chariotsolutions/meetupnotifier/EventIdRow.java]: I would add assertions (preconditions) in the constructor here that check the strings are not empty.  You can look into Google Guava's set of preconditions - https://code.google.com/p/guava-libraries/wiki/PreconditionsExplained

Comment (skapadia) [inline src/main/java/com/chariotsolutions/meetupnotifier/EventIdRow.java]: Yeah there's no need to use Derby, but thought you may be interested in knowing about it.  I don't know if I'd use either for serious production work, but fine for things like this.

File: src/main/java/com/chariotsolutions/meetupnotifier/MeetupNotifierImpl.java (hunk 2 above)
Comment (skapadia) [inline src/main/java/com/chariotsolutions/meetupnotifier/MeetupNotifierImpl.java]: In fact I would create a separate persistence / DAO class that handles retrieving and saving this data.  "DAO" = Data Access Object.  Its responsible for storing/retrieving calendar events to the DB.  It encapsulates how the data is written to the DB and what DB you write to.  So in the future that could change underneath without affecting anything using this class.

**Comment by skapadia** [comment]
LGTM.

**Pull request merged**